ext/droption/droption.h

```diff
@@ -570,6 +570,23 @@ template <typename T> class droption_t : public droption_parser_t {
         return true;
     }
 
+    static bool
+    is_negative(const std::string &s)
+    {
+        // Checks if the first non-space character of a string is a negative sign.
+        for (size_t i = 0; i < s.size(); i++) {
+            // XXX: The function isspace() only identifies ASCII whitespace characters. We
```

Comment:
isspace() checks for whitespace in the current locale and I believe if you set the locale to something like "en_US.UTF8" it will correctly return true for a unicode non-ascii whitespace -- are you seeing contradictory information?  Or are you saying that even if the locale is "C" or "POSIX" we'd prefer this to recognize a unicode non-ascii space -- but if the world has agreed that these locales should not be aware of non-ascii spaces then I think it's fine if we behave the same.

Reply:
Yes, the comment is contradictory. After I discussed this with @abhinav92003, he advised me to see the source code of strtoxx() and how it skips whitespaces. I found there are two solutions: 
(1) gcc uses ISSPACE() which is defined in [safe-ctype.h](https://github.com/gcc-mirror/gcc/blob/976196b3332d4f56248e7a1401e5e373501598e4/include/safe-ctype.h#L98) to skips only ASCII whitespace. 
(2) For [glibc](https://github.com/bminor/glibc/blob/a1bdd81664aa681364da368154c48501db249df9/stdlib/strtol_l.c#L150), [Apple libc](https://opensource.apple.com/source/Libc/Libc-167/stdlib.subproj/strtoul.c.auto.html), and [Android libc](https://android.googlesource.com/platform/bionic/+/a27d2baa/libc/stdlib/strtoul.c) directly use isspace() to identify whitespace and ignore the locale. 
And I didn't find how windows' strtoxx() skip whitespaces.

Do you agree that we keep the current implementation and add a comment that we ignore locale effects?

Reply:
In general the glibc routines, including isspace(), look at the locale, so the locale is not being ignored: so I don't follow your references to it ignoring the locale.  The issue is that isspace() takes as input just an unsigned char and so it can't handle a UTF-8 multi-byte char.  We would have to convert to UTF-16 and use iswspace() or std::isspace on wchar_t.
(Or maybe std::skipws handles UTF-8.)  This is more complex than I originally thought: while we have the code in DR to do the conversion it does not seem worth the effort at this point.  So yes I agree with keeping the current implementation, but please update the comment as this does not seem to be a locale issue but instead is the requirement that we convert to wchar_t to check one character.

Reply:
OK